build.ps1

```diff
@@ -473,15 +473,32 @@ function Build-CMakeProject {
       $cmakeGenerateArgs += @("-C", $CacheScript)
     }
     foreach ($Define in ($Defines.GetEnumerator() | Sort-Object Name)) {
-      # Avoid backslashes in defines since they are going into CMakeCache.txt,
-      # where they are interpreted as escapes. Assume all backslashes
-      # are path separators and can be turned into forward slashes.
-      $ValueWithPlaceholder = if ($SwiftSDK -ne "") { $Define.Value.Replace("$SwiftSDK", "<SDK>") } else { $Define.Value }
-      $ValueWithForwardSlashes = $ValueWithPlaceholder.Replace("\", "/")
-      if ($SwiftSDK -ne "") {
-        $ValueWithForwardSlashes = $ValueWithForwardSlashes.Replace("<SDK>", "\`"$SwiftSDK\`"")
+      # The quoting gets tricky to support defines containing compiler flags args,
+      # some of which can contain spaces, for example `-D` `Flags=-flag "C:/Program Files"`
+      # Avoid backslashes since they are going into CMakeCache.txt,
+      # where they are interpreted as escapes.
+      if ($Define.Value -is [string]) {
+        # Single token value, no need to quote spaces, the splat operator does the right thing.
+        $Value = $Define.Value.Replace("\", "/")
+      } else {
+        # Flags array, multiple tokens, quoting needed for tokens containing spaces
```

Comment:
I wish that there was a way to assert that it is a enumerable type (i.e. array).

Reply:
`-is [array]` works